README.md

```diff
@@ -193,6 +193,151 @@ spec:
+
+### Configuring Raw CNI
+Users can configure network attachment definition with CNI json as following options required:
+
+* `rawCNIConfig`: CNI JSON configuration for the network attachment
+
+```yaml
+spec:
+  additionalNetworks:
+  - name: test-network-1
+    namespace: namespace-test-1
+    rawCNIConfig: '{ "cniVersion": "0.3.1", "type": "macvlan", "master": "eth1", "mode": "bridge", "ipam": { "type": "dhcp" } }'
+    type: Raw
+```
+
+This config will generate the following network attachment definition:
+
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  # (snip)
+  name: test-network-1
+  namespace: namespace-test-1
+  ownerReferences:
+  - apiVersion: operator.openshift.io/v1
+    # (snip)
+spec:
+  config: '{ "cniVersion": "0.3.1", "type": "macvlan", "master": "eth1", "mode": "bridge", "ipam": { "type": "dhcp" } }'
+```
+
+### Configuring SimpleMacvlan
+SimpleMacvlan provides user to configure macvlan network attachments. macvlan creates a virtual copy of a master interface and assigns the copy a randomly generated MAC address. The pod can communicate with the network that is attached to the master interface. The distinct MAC address allows the pod to be identified by external network services like DHCP servers, firewalls, routers, etc. macvlan interfaces cannot communicate with the host via the macvlan interface. This is because traffic that is sent by the pod onto the macvlan interface is bypassing the master interface and is sent directly to the interfaces underlying network. Before traffic gets sent to the underlying network it can be evaluated within the macvlan driver, allowing it to communicate with all other pods that created their macvlan interface from the same master interface.
+
+Users can configure macvlan network attachment definition with following parameters, all of which are optional:
+
+* `master`: master is the host interface to create the macvlan interface from. If not specified, it will be default route interface
+* `mode`: mode is the macvlan mode: bridge, private, vepa, passthru. The default is bridge
+* `mtu`: mtu is the mtu to use for the macvlan interface. if unset, host's kernel will select the value
+* `ipamConfig`: IPAM (IP Address Management) configration: dhcp or static. The default is dhcp
+
+```yaml
+spec:
+  additionalNetworks:
+  - name: test-network-2
+    type: SimpleMacvlan
+    simpleMacvlanConfig:
+      master: eth0
+      mode: bridge
+      mtu: 1515
+      ipamConfig:
+        type: dhcp
+```
+
+#### Configuring Static IPAM
+Users can configure static IPAM with following parameters:
+
+* `addresses`:
+  * `address`: Address is the IP address in CIDR format, required
```

Comment:
Is the address required? I thought you could set that via an annotation?

Reply:
fixed.